controller/pod.go

```diff
@@ -104,23 +116,55 @@ type podConfig struct {
 }
 
 func addFunc(obj interface{}) {
+	if ns, ok := obj.(*v1.Namespace); ok {
+		lock.Lock()
+		isAmbient := false
+		if ns.Labels["istio.io/dataplane-mode"] == "ambient" {
+			isAmbient = true
+		}
+		nsMap[ns.Name] = isAmbient
+		if _, ok := podMap[ns.Name]; !ok {
+			podMap[ns.Name] = make(map[string]*v1.Pod)
+		}
+		lock.Unlock()
+		for _, pod := range podMap[ns.Name] {
+			addFunc(pod)
+		}
+	}
 	pod, ok := obj.(*v1.Pod)
 	if !ok || len(pod.Status.PodIP) == 0 {
 		return
 	}
-	if config.Mode == config.ModeIstio && !pods.IsIstioInjectedSidecar(pod) {
-		return
-	}
-	if config.Mode == config.ModeLinkerd && !pods.IsLinkerdInjectedSidecar(pod) {
-		return
-	}
-	if config.Mode == config.ModeKuma && !pods.IsKumaInjectedSidecar(pod) {
-		return
+	lock.Lock()
+	defer lock.Unlock()
+	if _, ok := podMap[pod.Namespace]; !ok {
+		podMap[pod.Namespace] = make(map[string]*v1.Pod)
+	}
+	podMap[pod.Namespace][pod.Name] = pod
+	isInjectedSidecar := false
+	switch config.Mode {
+	case config.ModeIstio:
+		isInjectedSidecar = pods.IsIstioInjectedSidecar(pod)
+	case config.ModeLinkerd:
+		isInjectedSidecar = pods.IsLinkerdInjectedSidecar(pod)
+	case config.ModeKuma:
+		isInjectedSidecar = pods.IsKumaInjectedSidecar(pod)
+	}
+	isAmbient := nsMap[pod.Namespace]
```

Comment:
This is not quite right:
https://github.com/istio/istio/blob/1.19.0/cni/pkg/ambient/podutil.go#L44